pyproject.toml

```diff
@@ -64,7 +65,12 @@ known-first-party = [
 ]
 
 [tool.mypy]
+exclude = ["micropip/_vendored/"]
 python_version = "3.12"
 show_error_codes = true
 warn_unreachable = true
 ignore_missing_imports = true
+
+[[tool.mypy.overrides]]
+module = "micropip._vendored.*"
+warn_unreachable = false
```

Comment:
I've had to keep this as there is some unreachable code in `src/packaging/`, which is not of concern for us, and TIL that the `warn_unreachable` option is still raised for excluded files.